### DISCUSSION TITLE
Link to GitRepo and commit changes on Update pannel

### DISCUSSION THREAD

**Discussion by TexZeTech** [mainsail-crew/mainsail#1840] [General]
So I asked this but clearly didn't word it correctly over on issue https://github.com/mainsail-crew/mainsail/issues/1481

I would like to see the following added to the interface and be usable even when there is no update available.

- A button ⓘ that will open the commit change page for that repo so a person can view that information even after an update.
- The version can link to the current version commit change page for that repo so a person can view that information even after an update.
- The Name will link to the github project.

**Comment by meteyou** [comment]
Please don't open different issues/posts for the identical thing. its easier for us to manage FR when they have only one conversation.

To be honest, a link to the repo itself would be possible, but i cannot open the "current release" like it is implemented with "open commits" right now. Moonraker doesn't support anything like that and i don't want to have the complete panel full of links. 

The next important thing is, that people should read the changes BEFORE they click on the update button. After updating something, its just to late, when something is broken.

**Reply by TexZeTech** [reply]
> Please don't open different issues/posts for the identical thing. its easier for us to manage FR when they have only one conversation.
> 

I understand you point, however you closed the original so in my mind that FR is closed until someone like you opens it again. 

I also though it was different enough that it warranted a discussion under ideas and to not waste time if its not a good idea. 

If others thought it was worth it then I would have opened a new FR and referenced the old one.

> 
> To be honest, a link to the repo itself would be possible, but i cannot open the "current release" like it is implemented with "open commits" right now. Moonraker doesn't support anything like that and i don't want to have the complete panel full of links.

That's fair and if it's a technical limitation then it is what it is.
But ultimately if you say no its no and I'm free to fork it and try and change it to my liking, I'm not trying to pick a fight I'm trying to ask for a feature while trying to not waist peoples time as I have more respect for individuals like yourself then most.

> 
> The next important thing is, that people should read the changes BEFORE they click on the update button. After updating something, its just to late, when something is broken.

Oh I agree 100% this is more of a convenience thing from my standpoint so I don't need to go to the repo page and manual look for it.

Hence the under Discussion -> ideas.

So do you want me to just close this and reopen the original FR and continue from there?

**Comment by meteyou** [comment]
> I understand you point, however you closed the original so in my mind that FR is closed until someone like you opens it again. 
> I also though it was different enough that it warranted a discussion under ideas and to not waste time if its not a good idea.
> If others thought it was worth it then I would have opened a new FR and referenced the old one.

The FR is only closed, you can just add posts to it and i will get an email notification. if there are valid reasons, i'll simply reopen the FR again. for me, an open FR just means that it's a TODO for me. that's why i try to keep it as clean as possible.

> That's fair and if it's a technical limitation then it is what it is.
> But ultimately if you say no its no and I'm free to fork it and try and change it to my liking, I'm not trying to pick a fight I'm trying to ask for a feature while trying to not waist peoples time as I have more respect for individuals like yourself then most.

Most of the software parts (klipper, mooonraker) don't have releases, so moonraker just update it to the current commit and moonraker only provides the "open commits". so i cannot display a dialog with "these are the commits of the newest release" in this case. for other software parts (like mainsail) it would only be possible to open a link to the current release, but i cannot display it directly in the webinterface.

so it wouldn't be possible to link to "each release", because it is just not "available". but a link to a repository, should be possible and would also help here.

i would just stay here until we have found a good idea. then we can just link the post in the FR and open it again.
